Sources/MongoSwift/BSON/Document.swift

```diff
@@ -328,6 +340,30 @@ extension Document {
             }
         }
     }
+
+#if swift(>=4.2)
+    /**
+     * Allows setting values and retrieving values using dot-notation syntax.
+     * For example:
+     *  ```
+     *  let d = Document()
+     *  d.a = 1
+     *  print(d.a) // prints 1
+     *  ```
+     * A nil return suggests that the key does not exist in the `Document`. A true BSON null is returned as
+     * an `NSNull`.
+     *
+     * Only available in Swift 4.2+.
```

Comment:
There is an [`@available` attribute](https://docs.swift.org/swift-book/ReferenceManual/Attributes.html#//apple_ref/doc/uid/TP40014097-CH35-ID347) in Swift that might be appropriate for specifying this kind of thing. I'm not totally sure what effect it has though.

Reply:
I added the `@available` tag instead of the `#if`. I don't believe it conditionally compiles, but it does provide a better compiler error if someone tries to use the dynamic lookup subscript on the wrong swift version. Since none of the code in the subscript is actually 4.2 specific, we don't need to conditionally compile anyways.

Reply:
awesome. does the tag mean it's available in 4.2 only, or will be there 4.2+?

Reply:
`@available(swift <version>)` is shorthand for `@available(swift, introduced: <version>)`, so it will work for 4.2+